File: android/src/main/kotlin/com/cloudwebrtc/webrtc/controller/ChannelNameGenerator.kt
package com.cloudwebrtc.webrtc.controller

/**
 * Generator for the all [io.flutter.plugin.common.MethodChannel] names created by `flutter_webrtc`.
 */
object ChannelNameGenerator {
  /** Prefix to prepend generated names with. */
  private const val PREFIX: String = "FlutterWebRtc"

  /**
   * Generates a new channel name for some controller by the provided ID.
   *
   * @param name Name of the controller to generate the channel name for (e.g. `PeerConnection`).
   * @param id Unique identifier of some concrete instance of some entity.
   *
   * @return Generated channel name.
   */
  fun name(name: String, id: Long): String {
    return "$PREFIX/$name/$id"
  }
}

File: ios/Classes/controller/ChannelNameGenerator.swift
/// Generator of names for Flutter method/event channels.
enum ChannelNameGenerator {
  /// Static prefix for all channels of this plugin
  private static let prefix: String = "FlutterWebRtc"

  /// Last generated ID.
  private static var lastId: Int = 0

  /// Returns a new ID for a channel.
  static func nextId() -> Int {
    ChannelNameGenerator.lastId += 1
    return self.lastId
  }

  /// Returns a name for a channel with the provided `name` and `id`.
  static func name(name: String, id: Int) -> String {
    "\(ChannelNameGenerator.prefix)/\(name)/\(id)"
  }
}
